include/NovelRT/Maths/GeoVector2F.h

```diff
@@ -21,157 +24,789 @@ namespace NovelRT::Maths
         {
             *reinterpret_cast<glm::vec2*>(this) =
                 *reinterpret_cast<const glm::vec2*>(this) * *reinterpret_cast<const glm::vec2*>(&other);
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the quotents of a Hadamard division using this and another
+         * GeoVector2F.
+         *
+         * @details
+         * An example of a Hadamard division of two vectors: \f[
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      \oslash
+         *      \begin{pmatrix}
+         *      2\\
+         *      3
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      \tfrac{8}{2}\\
+         *      \tfrac{6}{3}
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      4\\
+         *      2
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other vector who's component act as a divisor for this vector's components.
+         * @return A GeoVecor2F who's components consist of the entrywise quotent of this and another GeoVector2F.
+         */
         GeoVector2F operator/=(GeoVector2F other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) =
                 *reinterpret_cast<const glm::vec2*>(this) / *reinterpret_cast<const glm::vec2*>(&other);
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise sum of this and another scalar and applies it to
+         * this instance.
+         *
+         * @details
+         * An example of summing up a vector and a scalar: \f[
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      +
+         *      6
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      +
+         *      \begin{pmatrix}
+         *      6\\
+         *      6
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      8 + 6 \\
+         *      6 + 6
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      14\\
+         *      12
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other scalar that gets added up with each component of this vector.
+         * @return This GeoVecor2F who's components consist of the entrywise sum of this and another scalar.
+         */
         inline GeoVector2F operator+=(float other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) = *reinterpret_cast<const glm::vec2*>(this) + other;
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise difference of this and another scalar and applies it
```

Comment:
construct on operator. Also this is in-place memory overwriting as it returns `T&` to `this`.